src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -4079,6 +4079,14 @@ def aks_agentpool_upgrade(cmd, client, resource_group_name, cluster_name,
             'node version please use the "--node-image-only" option only.'
         )
 
+    # Note: we exclude this option because node image upgrade can't accept nodepool put fields like max surge
+    if max_surge and node_image_only:
+        raise CLIError(
```

Comment:
I think replacing `CLIError` with `MutuallyExclusiveArgumentError` can more clearly point out the cause of the error.

Reply:
Updated. There is another error directly above that is listed as CLIError. However, is a mutually exclusive argument check between kubernetes_version and node_image_only.

Do you want that updated as well, so better to not change the existing error in this PR?

Reply:
Also, update the help message. Unsure if that is desired or not, but felt that this mutually exclusive issue was less intuitive

Reply:
`CLIError` is an error type that has been deprecated. Although simple and straightforward, it is too general, and it is not possible to distinguish the cause of the error from the user, the client or the server. You can refer to the recommended error types defined in this file [azclierror.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py).

As for other places where `CLIError` is used, if it is not modified in this PR, just ignore it. I will gradually update these inappropriate error types.

More accurate error types can also help the telemetry and our azcli aks runner/check-in pipleine to determine whether the error encountered is a real issue :smiley: